pkg/build/registry/buildconfig/webhook.go

```diff
@@ -72,9 +73,13 @@ func (w *WebHook) ServeHTTP(writer http.ResponseWriter, req *http.Request, ctx k
 		Revision:    revision,
 		Env:         envvars,
 	}
-	if _, err := w.instantiator.Instantiate(config.Namespace, request); err != nil {
+	newBuild, err := w.instantiator.Instantiate(config.Namespace, request)
+	if err != nil {
 		return errors.NewInternalError(fmt.Errorf("could not generate a build: %v", err))
 	}
+	// Send back the build name so that the client can alert the user.
+	newBuildJson, _ := json.Marshal(newBuild)
```

Comment:
it's not really "dying", because the build is going to get run regardless, it's about what you return to the caller.  

Certainly just proceeding w/ writing the json seems like a mistake since you have no idea what's in there, given that the marshal returned an error.